src/module_wrap.cc

```diff
@@ -884,16 +887,96 @@ Maybe<URL> ResolveExportsTarget(Environment* env,
   return Just(subpath_resolved);
 }
 
+Maybe<URL> ResolveExportsTarget(Environment* env,
+                                const URL& pjson_url,
+                                Local<Value> target,
+                                const std::string& subpath,
+                                const std::string& pkg_subpath,
+                                const URL& base,
+                                bool throw_invalid = true) {
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+  if (target->IsString()) {
+    Utf8Value target_utf8(isolate, target.As<v8::String>());
+    std::string target_str(*target_utf8, target_utf8.length());
+    Maybe<URL> resolved = ResolveExportsTargetString(env, target_str, subpath,
+        pkg_subpath, pjson_url, base, throw_invalid);
+    if (resolved.IsNothing()) {
+      return Nothing<URL>();
+    }
+    return FinalizeResolution(env, resolved.FromJust(), base);
+  } else if (target->IsArray()) {
+      Local<Array> target_arr = target.As<Array>();
+      const uint32_t length = target_arr->Length();
+      if (length == 0) {
+        if (throw_invalid) {
+          ThrowExportsInvalid(env, pkg_subpath, target, pjson_url, base);
+        }
+        return Nothing<URL>();
+      }
+      for (uint32_t i = 0; i < length; i++) {
+        auto target_item = target_arr->Get(context, i).ToLocalChecked();
+        if (!target_item->IsArray()) {
+          Maybe<URL> resolved = ResolveExportsTarget(env, pjson_url,
+              target_item, subpath, pkg_subpath, base, false);
+          if (resolved.IsNothing()) continue;
+          return FinalizeResolution(env, resolved.FromJust(), base);
+        }
+      }
+      if (throw_invalid) {
+        auto invalid = target_arr->Get(context, length - 1).ToLocalChecked();
+        Maybe<URL> resolved = ResolveExportsTarget(env, pjson_url, invalid,
+            subpath, pkg_subpath, base, true);
+        CHECK(resolved.IsNothing());
+      }
+      return Nothing<URL>();
+  } else if (target->IsObject() && !target->IsNull()) {
```

Comment:
```suggestion
  } else if (target->IsObject()) {
```

(null is not considered an object by the C++ API, as it should be :wink:)

Reply:
TIL! Thanks.